### ISSUE TITLE
atom.showSaveDialogSync broken in version 1.15.0

### ISSUE THREAD

**Issue by cs150bf** [atom/atom#14109]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

`atom.showSaveDialogSync('some/chosen/path/to/file')` used to work (still worked in version 1.14.0) but doesn't work in version 1.15.0.

### Steps to Reproduce

1. Open Atom in debug mode
2. Open console panel
3. Create a new file and edit something in the editor
4. In the console panel, type `atom.showSaveDialogSync('some/chosen/path/filename')`

**Expected behavior:** A save dialog pops out with the filename and path pre-set according to the param. 

**Actual behavior:** [What actually happens] The save dialog doesn't set to designated path and filename.

**Reproduces how often:** Always in version 1.15.0.

### Versions

Atom    : 1.15.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0

### Additional Information

Another example: 

This function is used in the Markdown Preview package (https://github.com/atom/markdown-preview/blob/master/lib/markdown-preview-view.coffee#L292). When the markdown preview pane is opened, right clicking on the preview and selecting "Save as HTML" used to lead to a popped out save dialog with the "Save As..." field auto-filled with a filename ending with ".html". But in version 1.15.0, the filename is just "Untitled".

**Comment by cs150bf** [comment]
## My suspect: 
I'm looking at https://github.com/atom/atom/commit/c6cae5b8fdfea982084b068a0b38a7c988c52f51 and I have a feeling that all those change related to `showSaveDialog` have to do with this problem.

## I traced this issue as follows (links to blame page):

`atom-environment.coffee (showSaveDialogSync)` --> `application-delegate.coffee (showSaveDialog)` [#156](https://github.com/atom/atom/blame/b32f409a20b7396a13830f202ce1591c27442ab9/src/application-delegate.coffee#L156) --> `atom-window.coffee (showSaveDialog)` [#294](https://github.com/atom/atom/blame/1ff5c9e684a6c53ef460a2aea5ac931b0b6dd137/src/main-process/atom-window.coffee#L294)

(Please note the changes between 1.14.0 and 1.15.0)

## My Guess:
Should it be `const {dialog} = require('electron').remote` in [atom-window.cofee](https://github.com/atom/atom/blob/1ff5c9e684a6c53ef460a2aea5ac931b0b6dd137/src/main-process/atom-window.coffee)? Because I see that in the **old** (_as in, [v1.14.0](https://github.com/atom/atom/commit/e1fe9eae19fb893b531556de47c8b3d88cec1dc6)_) `application-delegate.coffee` [#158](https://github.com/atom/atom/blame/ae9d949ee4e199d1884bed89083196fa375bbd34/src/application-delegate.coffee#L158), the function `dialog.showSaveDialog` is called with:
```javascript
remote.dialog.showSaveDialog remote.getCurrentWindow(), params
```
But now the function `dialog.showSaveDialog` is called in `atom-window.coffee`[#1](https://github.com/atom/atom/blame/1ff5c9e684a6c53ef460a2aea5ac931b0b6dd137/src/main-process/atom-window.coffee#L1) and [#299](https://github.com/atom/atom/blame/1ff5c9e684a6c53ef460a2aea5ac931b0b6dd137/src/main-process/atom-window.coffee#L299) with
```javascript
{BrowserWindow, app, dialog, ipcMain} = require 'electron'
.
.
.
dialog.showSaveDialog(this, params)
```

## Same issue?

I think the issue https://github.com/atom/atom/issues/13978 is caused by the malfunctioning of `showSaveDialog` reported here.

**Comment by nathansobo** [comment]
@cs150bf Thanks for taking care of this. @50Wliu Should we backport the fix at least to 1.16 so it goes out next week?
